### DISCUSSION TITLE
Modbus TCP in CPP?

### DISCUSSION THREAD

**Discussion by parkykk** [alisitsyn/modbus_support#4] [Q&A]
Hi @alisitsyn 

Do you have any plan to provide Modbus TCP Slave/Master examples in ccp? 

The examples(which are written in C) in the github esp-idf are working fantastic! I have been working for a couple of weeks to covert the C code to cpp, but no luck. My knowledge is not good enough. 

Could you please provide any advice?

Kind regards 
parkykk

**Comment by alisitsyn** [comment]
Hi @parkykk ,

Thank you for your interest and sorry for the late response. Please take a look to the CPP wrapper code [here](https://github.com/alisitsyn/modbus_support/blob/mb_support/modbus_add_cpp_wrapper_v2). This code uses the wrapper for RTU now and I plan to provide examples of using the TCP as well. I will inform you as soon as possible. 

Best regards,
alisitsyn
